tasks/main.yml

```diff
@@ -0,0 +1,162 @@
+---
+- name: Install role requirements
+  import_tasks: requirements.yml
+  when: install_home_assistant_supervised_ansible_requirements|bool
+
+- name: Install OS package prereqs
+  apt:
+    name:
+      - systemd
+      - network-manager
+      - apparmor
+      - jq
+      - curl
+      - dbus
+    state: latest
+    update_cache: yes 
+    
+- name: Disable Modem Manager to prevent issues when using serial devices
+  service:
+    name: ModemManager
+    enabled: no
+    state: stopped
+
+- name: Create Docker config directory 
+  file:
+    path: /etc/docker 
+    state: directory
+
+- name: Set up docker config file
+  copy:
+    src: docker_daemon.json
+    dest: /etc/docker/daemon.json
+  notify:
+    - restart docker
+
+- name: Fix kernel dmesg restriction 
+  sysctl:
+    name: kernel.dmesg_restrict 
+    value: '0'
+    sysctl_set: yes
+    state: present
+    reload: yes
+
+- name: Create config for NetworkManager   
+  copy:
+    src: "{{ networkmanager_file['source'] }}" 
+    dest: "{{ networkmanager_file['target'] }}" 
+  loop:
+    - source: NetworkManager.conf
+      target: /etc/NetworkManager/NetworkManager.conf   
+    - source: system-connection-default 
+      target: /etc/NetworkManager/system-connections/default
+  loop_control:
+    loop_var: networkmanager_file
+    label: "{{ networkmanager_file['source'] }}"
+  notify:
+    - restart networkmanager
+
+- name: Overwrite network interfaces file
+  copy:
+    src: interfaces
+    dest: /etc/network/interfaces
+  notify:
+    - restart networkmanager
+ 
+- name: Init folders
+  file:
+    path: "{{ data_share }}"  
+    state: directory
+
+- name: Read version infos from web
+  uri:
+    body_format: json
+    url: https://version.home-assistant.io/stable.json
+  register: hassio_stable_out
+  
+- name: Set hassio_version fact 
+  set_fact:
+    hassio_version: "{{ hassio_stable_out['json']['supervisor'] }}"
+  
+- name: Write config
```

Comment:
From a formatting perspective, do the "Read Version infos from web" and "set hassio_version fact" tasks feed any info into "write config" or "ensure all changed service are restarted"?

I don't see any dependencies untl "Install supervisor Docker container" - if that's the case, might make sense to just switch the order of these tasks. If any issues and need to debug, the tasks that pull `hassio_version` var will be right before the task using it

Reply:
Yeah, that's the order they have it in the script, but I think it makes enough sense to move it and it doesn't change any of the functionality